res/res_stir_shaken/crypto_utils.h

```diff
@@ -0,0 +1,264 @@
+/*
+ * Asterisk -- An open source telephony toolkit.
+ *
+ * Copyright (C) 2023, Sangoma Technologies Corporation
+ *
+ * George Joseph <gjoseph@sangoma.com>
+ *
+ * See http://www.asterisk.org for more information about
+ * the Asterisk project. Please do not directly contact
+ * any of the maintainers of this project for assistance;
+ * the project provides a web site, mailing lists and IRC
+ * channels for your use.
+ *
+ * This program is free software, distributed under the terms of
+ * the GNU General Public License Version 2. See the LICENSE file
+ * at the top of the source tree.
+ */
+#ifndef _CRYPTO_UTILS_H
+#define _CRYPTO_UTILS_H
+
+#include "openssl/x509.h"
+#include "openssl/x509_vfy.h"
+
+#include "asterisk.h"
+#include "asterisk/logger.h"
+#include "asterisk/stringfields.h"
+
+struct ast_X509_Extension {
+	AST_DECLARE_STRING_FIELDS(
+		AST_STRING_FIELD(oid);
+		AST_STRING_FIELD(short_name);
+		AST_STRING_FIELD(long_name);
+	);
+	int nid;
+};
+
+/*!
+ * \brief Print a log message with any OpenSSL errors appended
+ *
+ * \param level	Type of log event
+ * \param file	Will be provided by the AST_LOG_* macro
+ * \param line	Will be provided by the AST_LOG_* macro
+ * \param function	Will be provided by the AST_LOG_* macro
+ * \param fmt	This is what is important.  The format is the same as your favorite breed of printf.  You know how that works, right? :-)
+ */
+void ast_log_openssl(int level, char *file, int line,
```

Comment:
By naming these with `ast_` in the front will they be exported? If so, do we want that? If not, should we rename them to avoid confusion?

Reply:
Yeah, these were all left over from when this was a separate module.  I've now removed the ast_ prefix.